core/src/com/unciv/ui/worldscreen/mainmenu/DropBox.kt

```diff
@@ -219,4 +221,77 @@ class LockFile {
     var lockData = UUID.randomUUID().toString()
 }
 
-class DropBoxFileConflictException: Exception()
\ No newline at end of file
+/**
+ *	Wrapper around OnlineMultiplayer's synchronization facilities.
+ *
+ *	Based on the design of Mutex from kotlinx.coroutines.sync, except that when it blocks,
+ *	  it blocks the entire thread for an increasing period of time via Thread.sleep()
+ */
+class ServerMutex(val gameInfo: GameInfo) {
+	var locked = false
+
+    /**
+     * Try to obtain the server lock ONCE
+     * @see OnlineMultiplayer.tryLockGame
+     * @see lock
+     * @see unlock
+     * @return true if lock is acquired
+     */
+	fun tryLock(): Boolean {
+		locked = OnlineMultiplayer().tryLockGame(gameInfo.asPreview())
```

Comment:
Should we move the OnlineMultiplayer calls into the ServerMutex? I mean its called only once by ServerMutex and having two ways to lock is probably just confusing so we could move the logic from OnlineMultiplayer to ServerMutex

Reply:
I considered that as well. I wasn't sure if you had a particular design in mind when you created those calls, but if they can be moved that's fine too. I'll take care of it.